### ISSUE TITLE
[java] Disallow try-with-resources block around an `ExecutorService.execute` call

### ISSUE THREAD

**Issue by boris-petrov** [pmd/pmd#5306]
**Proposed Rule Name:**

No idea, I'm not good with names. :smile: 

**Proposed Category:** One of [Best Practices | Code Style | Design | Documentation | Error Prone | Multithreading | Performance | Security]

Probably `Multithreading`?

**Description:**

See the code sample below:

**Code Sample:** This should include code, that should be flagged by the rule. If possible, the "correct" code
according to this new rule should also be demonstrated.

```java
class AutoCloseableSemaphore implements AutoCloseable {
  ...
}

var executor = Executors.newVirtualThreadPerTaskExecutor();
var lock = new AutoCloseableSemaphore(10);
for (var i = 0; i < 100; i++) {
  try (lock) {
    executor.execute(() => ...);
  }
}
```

I've done that mistake a few times already and it's really annoying. One might think that this code will execute at most 10 things at the same time... but it will actually run all 100 in parallel. So I suggest a rule that checks for usage of `try-with-resources` around `ExecutorService.execute` methods (and the likes) as that's most likely a mistake by the programmer - the `execute` method will return immediately and the resource is freed (probably before even the callback has began executing).

**Possible Properties:**

*   Should this rule be customizable via properties?
No
